db-engine/src/main/java/com/splicemachine/db/iapi/types/DataValueFactoryImpl.java

```diff
@@ -799,13 +799,29 @@ public StringDataValue         getNullChar(StringDataValue previous,
         {
             if (collationType == StringDataValue.COLLATION_TYPE_UCS_BASIC)
                 return getNullChar(previous);
-            
+
             if (previous == null)
                 return new CollatorSQLChar(getCharacterCollator(collationType));
-            
+
             previous.setToNull();
             return previous;
-         }
+        }
+
+        public StringDataValue getNullChar(StringDataValue previous, int collationType, int maxSize)
+        throws StandardException
+        {
+            SQLChar ret = null;
+
+            if (collationType == StringDataValue.COLLATION_TYPE_UCS_BASIC) {
+                ret = (SQLChar) getCharDataValue((String) null);
```

Comment:
Why this path does not take into consideration the passed in argument _previous_ if it is not null? (I'm comparing it with the existing function getNullChar(StringDataValue, int).

Reply:
I made the logic a bit cleaner, the behavior is the same. We return a NULL value, always. If `previous` is defined, then the `previous` is also set to NULL.

Reply:
My understanding is that if _previous_ is not null, we want to reuse it. But with this function, we always create a new instance. Am I wrong?

Reply:
Yes, you are right. I fixed the logic of using previous value.